File: src/main/java/com/in3/task/model/SearchCriteriaModel.java
package com.in3.task.model;

import java.util.ArrayList;

public class SearchCriteriaModel {
    private ArrayList<String> ram = new ArrayList<>();
    private ArrayList<String> cpu = new ArrayList<>();
    private ArrayList<String> os = new ArrayList<>();
    private ArrayList<String> inches = new ArrayList<>();

    public ArrayList<String> getRam() {
        return ram;
    }

    public void setRam(ArrayList<String> ram) {
        this.ram = ram;
    }

    public ArrayList<String> getCpu() {
        return cpu;
    }

    public void setCpu(ArrayList<String> cpu) {
        this.cpu = cpu;
    }

    public ArrayList<String> getOs() {
        return os;
    }

    public void setOs(ArrayList<String> os) {
        this.os = os;
    }

    public ArrayList<String> getInches() {
        return inches;
    }

    public void setInches(ArrayList<String> inches) {
        this.inches = inches;
    }
}

File: src/main/resources/webapp/src/app/model/search.criteria.model.ts
export class SearchCriteriaModel {
  private _os: string[];

  get os(): string[] {
    return this._os;
  }

  set os(value: string[]) {
    this._os = value;
  }

  private _ram: string[];

  get ram(): string[] {
    return this._ram;
  }

  set ram(value: string[]) {
    this._ram = value;
  }

  private _cpu: string[];

  get cpu(): string[] {
    return this._cpu;
  }

  set cpu(value: string[]) {
    this._cpu = value;
  }

  private _inches: string[];

  get inches(): string[] {
    return this._inches;
  }

  set inches(value: string[]) {
    this._inches = value;
  }
}
